src/java/org/apache/cassandra/service/accord/AccordJournalTable.java

```diff
@@ -492,15 +489,15 @@ public K key()
         }
 
         @Override
-        public void readAllForKey(K key, RecordConsumer<K> reader)
+        public void readAllForKey(K key, RecordConsumer<K> reader, boolean includeUpdates)
```

Comment:
`includeUpdates` seems to imply normally we don't, whereas here we're _only_ reading updates right?

Perhaps it would be clearer to accept a segment id or position we read from? 

Are we also confident this is fine with compaction? Can anything weird happen if new data is compacted with old data?

Reply:
I think compaction might cause problems actually, since we aren't merging records here I think? We're just loading new records then earlier ones. But we need to really merge on read, as I think we treat them all as newer than the records we had on startup, even though we could have intermixed older data?

Reply:
Hrm, or in fact it looks like we aren't even looking at sstable segments. So, are we disabling the compactor that flushes to journal until this completes? if so it might be safe, but we would need to make sure we can guarantee this doesn't get started until we finish as it will silently corrupt things.